<!--
  Copyright 2016 The Cartographer Authors

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
-->

<launch>
  <arg name="state_filename" default=""/>
  <arg name="save_state_filename" default="/home/stn/slam/trajectories/carto_loc_test_240_0.2h/saved.pbstream"/>
  <arg name="bag_filename" default="/home/stn/Dataset/other/carto_loc_test_240_0.2h.bag"/>
  <arg name="traj_filename" default=""/>
  <arg name="traj_id" default="0"/>
  <param name="/use_sim_time" value="true" />
  <param name="robot_description"
    textfile="$(find cartographer_ros)/urdf/backpack_2d.urdf" />

  <node name="robot_state_publisher" pkg="robot_state_publisher"
    type="robot_state_publisher" />

  <node name="tf_listen" pkg="cartographer_ros" type="tf_listen" output="screen">
    <param name="traj_filename" value="$(arg traj_filename)"/>
    <param name="traj_id" value="$(arg traj_id)"/>
  </node>

  <node name="cartographer_node" pkg="cartographer_ros"
      type="cartographer_node" args="
          -configuration_directory $(find cartographer_ros)/configuration_files
          -configuration_basename my_2d_localization.lua
          -save_state_filename $(arg save_state_filename)
          -load_state_filename $(arg state_filename)"
      output="screen">
  </node>

  <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
      type="cartographer_occupancy_grid_node" args="-resolution 0.05" />

  <node name="rviz" pkg="rviz" type="rviz" required="true"
      args="-d $(find cartographer_ros)/configuration_files/demo_2d.rviz" />
  <node name="playbag" pkg="rosbag" type="play" required="true"
      args="--clock $(arg bag_filename)" />
</launch>
